app/views/tags/index.html.erb

```diff
@@ -21,8 +21,23 @@
 
 <main id="main-content" class="crayons-layout crayons-layout--header-inside">
   <header class="crayons-page-header">
-    <h1 class="crayons-title"><%= t("views.tags.heading") %></h1>
-    <%= link_to t("views.tags.following_tags"), dashboard_following_tags_path, class: "crayons-btn crayons-btn--ghost" %>
+    <h1 class="crayons-title">
+      <% if params[:q].present? %>
+        <%= t("views.tags.heading.search_results", q: params[:q]) %>
+      <% else %>
+        <%= t("views.tags.heading.default") %>
+      <% end %>
+    </h1>
+    <form accept-charset="UTF-8" method="get" action="<%= tags_path %>" role="search">
+      <div class="crayons-fields crayons-fields--horizontal">
+        <div class="crayons-field flex-1 relative">
+          <input class="crayons-header--search-input crayons-textfield js-search-input" type="text" id="nav-search" name="q" placeholder="<%= t("views.tags.search.placeholder") %>" autocomplete="off" />
+          <button type="submit" aria-label="<%= t("views.search.icon.aria_label") %>" class="c-btn c-btn--icon-alone absolute inset-px left-auto mt-0 py-0">
```

Comment:
Another wee nitpick - it would be great if we could create a separate aria-label i18n string for this, otherwise screen reader users can't differentiate easily between the general search and the tags search. Something like "search tags" would work well

Reply:
added here too! 😄